### PR TITLE
Preparations for d.py 1.4 (includes breaking changes related to mass mentions)

### LINKED ISSUE #4069: Alias does not handle space-only alias gracefully
# Command bugs

#### Command name

`[p]alias add` and `[p]alias global add`

#### What cog is this command from?

Alias

#### What were you expecting to happen?

My bot to say `You attempted to create a new alias with a space but that name is an invalid alias name. Alias names may not contain spaces.`, or something close to that, handling it gracefully.

#### What actually happened?

Raised an error

> Exception in command 'alias add'
> Traceback (most recent call last):
>   File "/home/neuroassassin/Desktop/Red/modok/lib/python3.8/site-packages/discord/ext/commands/core.py", line 83, in wrapped
>     ret = await coro(*args, **kwargs)
>   File "/home/neuroassassin/Desktop/Red/modok/lib/python3.8/site-packages/redbot/cogs/alias/alias.py", line 142, in _add_alias
>     is_command = self.is_command(alias_name)
>   File "/home/neuroassassin/Desktop/Red/modok/lib/python3.8/site-packages/redbot/cogs/alias/alias.py", line 64, in is_command
>     command = self.bot.get_command(alias_name)
>   File "/home/neuroassassin/Desktop/Red/modok/lib/python3.8/site-packages/redbot/core/bot.py", line 196, in get_command
>     com = super().get_command(name)
>   File "/home/neuroassassin/Desktop/Red/modok/lib/python3.8/site-packages/discord/ext/commands/core.py", line 1144, in get_command
>     obj = self.all_commands.get(names[0])
> IndexError: list index out of range
> 

#### How can we reproduce this issue?

`[p]alias add " " ping` or `[p]alias global add " " ping`.  Must be only the space.

### PR DESCRIPTION
**Pull request by Jackenmen**
### Type

- [ ] Bugfix
- [x] Enhancement
- [ ] New feature

### Description of the changes
This PR contains changes needed due to d.py 1.4 release as well as this (breaking) change:
> 1. By default, none of the `.send()` methods will mention roles or @​everyone/@​here.
> 1.1. d.py 1.4 introduces allowed_mentions kwarg to `.send()` methods that can be used to make the message mention roles or @​everyone/@​here (NOTICE: this may still change before d.py 1.4 release)
> 1.2. `sanitize_roles` kwarg mentioned in the previous breaking change notice from Flame will NOT be implemented.
> 1.3. The default value of the `filter` kwarg in `Context.send()` will be changed to `None` which means that enabling @​everyone/@​here mentions will now require using the `allowed_mentions` kwarg instead.

Fixes #4069 (indirectly, through dependency update)

### CODE REVIEW

**Comment by Jackenmen** [comment]
A reminder to myself that I need to look through `DEP-WARN` comments again before marking this as ready for review.

**Comment by Jackenmen** [comment]
This is now ready for review.

File: setup.cfg [hunk 1]
```diff
@@ -44,7 +44,7 @@ install_requires =
     click==7.1.2
     colorama==0.4.3
     contextlib2==0.6.0.post1
-    discord.py==1.3.4
+    discord.py==1.4.1
```
Comment (Drapersniper) [inline setup.cfg:47]: can you also bump
`Red-Lavalink==0.5.1` to 0.6.0 ?

**Review by Drapersniper — APPROVED** [review]
LGTM, thanks @jack1142 and @PredaaA

**Pull request merged**